tests/test_project.py

```diff
@@ -1131,4 +1141,4 @@ def test_dag(self):
                        [0, 0, 0, 0, 0, 0, 0],
                        [0, 0, 0, 0, 0, 0, 0]]
 
-        assert adj == adj_correct
```

Comment:
All files should end with a newline character. It seems your editor may have trimmed this, or you removed it manually. Please revert this change.